### ISSUE TITLE
How can I adapt to a regression problem?

### ISSUE THREAD

**Issue by ywy9876** [tkipf/gcn#80]
Hello,
if my task was to "compute", for exmaple, the pagerank of the nodes (or other metrics of node), using GNN (to see if it's possible). Which changes should I make in order to achieve it?  I think the input labels should be N*1 dimension now as it's a single value of the metric.  And what about the _loss and masked_softmax_cross_entropy function, what corresponding should I make?

Best regards.

**Comment by tkipf** [comment]
PageRank can be seen as a particular instance of a graph neural net, but
with fixed, pre-defined weights and message functions. You can probably
train a GCN or graph neural net to predict page rank scores as a regression
problem (given ground truth pagerank scores), but this somewhat defeats the
purpose (you could just use the PageRank algorithm altogether instead).

**Comment by ywy9876** [comment]
Thanks for the advise. 
In our side-project we wanna see if embeddings are transferable accross different betweenness or other types of centrality measures. So it would be nice to know how to adapt the code part to meet the condition.
